float8_experimental/float8_ops.py

```diff
@@ -140,3 +141,29 @@ def autocast_to_copy(aten_op, args, kwargs=None):
     return Float8Tensor(
         args[0]._data, args[0]._scale, kwargs["dtype"], args[0]._emulate
     )
+
+
+@implements([c10d_functional.all_gather_into_tensor.default])
+def allgather_fp8(aten_op, args, kwargs=None):
+    """
+    override funcol with FP8 handling
+    """
+    fp8_input = args[0]
+    assert isinstance(fp8_input, Float8Tensor)
+
+    fp8_data = fp8_input._data
+    fp8_data = fp8_data.view(torch.int8)
```

Comment:
curious q: is there any difference between this and torch.uint8?

Reply:
JFYI @wanchaol : I was using `uint8` for fp8 all-gather work because `uint8` felt more traditional for "the raw bytes", but I think it should not matter much.

Reply:
I was following https://github.com/pytorch-labs/float8_experimental/blob/main/float8_experimental/distributed_utils.py#L49 but torch.uint8 make more sense, switching over